doc/_scripts/gen_devicetree_rest.py

```diff
@@ -500,6 +554,20 @@ def print_binding_page(binding, base_names, vnd_lookup, dup_compats,
           f':ref:`{vnd_lookup.vendor(vnd)} <{vnd_lookup.target(vnd)}>`\n',
           file=string_io)
 
+    # Link to driver implementation (if it exists).
+    compatible = compatible.replace(",", "_")
+    compatible = compatible.replace("-", "_")
```

Comment:
`scripts/dts/gen_defines.py` uses the following:

```python
def str2ident(s: str) -> str:
    # Converts 's' to a form suitable for (part of) an identifier

    return re.sub('[-,.@/+]', '_', s.lower())

```

Reply:
Ah! Thanks! I looked for something like this but gave up too early, apparently :)

Reply:
Fixed, but I am not sure about all those special characters in addition to `-` and `,` tbh... Like, can a compatible really contain a forward slash?

Reply:
I don't know either, but consistency is a valid argument. Maybe the helper function could be moved up a layer.